ci/driver.sh

```diff
@@ -0,0 +1,107 @@
+#!/bin/bash --login
+
+my_dir="$( cd "$( dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd )"
+
+# ==============================================================================
+usage() {
+  set +x
+  echo
+  echo "Usage: $0 -t <target> -h"
+  echo
+  echo "  -t  target/machine script is running on    DEFAULT: $(hostname)"
+  echo "  -h  display this message and quit"
+  echo
+  exit 1
+}
+
+# ==============================================================================
+# First, set up runtime environment
+
+export TARGET="$(hostname)"
+
+while getopts "t:h" opt; do
+  case $opt in
+    t)
+      TARGET=$OPTARG
+      ;;
+    h|\?|:)
+      usage
+      ;;
+  esac
+done
+
+case ${TARGET} in
+  hera | orion)
+    echo "Running Automated Testing on $TARGET"
+    source $MODULESHOME/init/sh
+    source $my_dir/${TARGET}.sh
+    module purge
+    module use $RDAS_MODULE_USE
+    module load RDAS/$TARGET
+    module list
+    ;;
+  *)
+    echo "Unsupported platform. Exiting with error."
+    exit 1
+    ;;
+esac
+
+# ==============================================================================
+# pull on the repo and get list of open PRs
+cd $RDAS_CI_ROOT/repo
+CI_LABEL="${RDAS_CI_HOST}-RT"
+gh pr list --label "$CI_LABEL" --state "open" | awk '{print $1;}' > $RDAS_CI_ROOT/open_pr_list
+open_pr_list=$(cat $RDAS_CI_ROOT/open_pr_list)
+
+# ==============================================================================
+# clone, checkout, build, test, etc.
+repo_url="https://github.com/NOAA-EMC/RDASApp.git"
+# loop through all open PRs
+for pr in $open_pr_list; do
+  gh pr edit $pr --remove-label $CI_LABEL --add-label ${CI_LABEL}-Running
+  echo "Processing Pull Request #${pr}"
+  mkdir -p $RDAS_CI_ROOT/PR/$pr
+  cd $RDAS_CI_ROOT/PR/$pr
+
+  # clone copy of repo
+  git clone --recursive $repo_url
```

Comment:
@CoryMartin-NOAA  Will use "--recurse-submodules"  better?   This form of git clone I just learned from Donnie , which combines "recursive " and following "git submot init and update"

Reply:
You don't necessarily want to do that, that won't give you the expected versions/tags of the submodules but rather the newest, right?

Reply:
"Git submodule --update" will get the submodules the same as what is cloned.  Is this what we expect? 
Otherwise, the submodules are maybe different from what are in the branch/PR we are cloning out.  
So if I understand right, we should always run those submodule updates following a clone .

Reply:
Ting, I think I see your point. But the submodule update line may need to go after the `gh pr checkout` command, to update the submodules for that PR.